### PR TITLE
luks_device: allow to provide passphrases base64-encoded

### LINKED ISSUE #827: luks_device: can't use binary data for passphrase
##### SUMMARY

I have a partition that is encrypted using a randomly generated keyfile, which I store in a vault. I'm trying to write a playbook that opens the encrypted partition using the keyfile without uploading the keyfile to the managed node. Note that running
```
cat keyfile | ssh -p 2222 root@localhost cryptsetup open --key-file - /dev/vda3 data
```
works (where keyfile is the unencrypted keyfile).

Using `ansible.builtin.command` and `ansible.builtin.shell` also doesn't work (see ansible/ansible#84497)

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

luks_device

##### ANSIBLE VERSION

```paste below
ansible [core 2.18.1]
  config file = /home/ilia/[...]/ansible.cfg
  configured module search path = ['/home/ilia/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.13/site-packages/ansible
  ansible collection location = /home/ilia/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.13.1 (main, Dec  4 2024, 18:05:56) [GCC 14.2.1 20240910] (/usr/bin/python)
  jinja version = 3.1.5
  libyaml = True
```

##### COLLECTION VERSION

```paste below
# /usr/lib/python3.13/site-packages/ansible_collections
Collection       Version
---------------- -------
community.crypto 2.22.3
```

##### CONFIGURATION

```paste below
CONFIG_FILE() = /home/ilia/[...]/ansible.cfg
DEFAULT_HOST_LIST(/home/ilia/[...]/ansible.cfg) = ['/home/ilia/[...]/inventory.yml']
DEFAULT_VAULT_ID_MATCH(/home/ilia/[...]/ansible.cfg) = true
EDITOR(env: EDITOR) = vim

GALAXY_SERVERS:
```

##### OS / ENVIRONMENT

Control: Arch Linux
Managed node: Debian 12

##### STEPS TO REPRODUCE

```yaml
- name: unlock encrypted data partitions
  hosts: all
  vars:
    keyfile: "{{ lookup('ansible.builtin.unvault', 'keyfiles/' ~ inventory_hostname) }}"
  tasks:
    - ansible.builtin.debug:
        msg: "{{ keyfile }}"
    - name: unlock data partitions
      community.crypto.luks_device:
        device: "{{ item.device }}"
        name: "{{ item.mountpoint | basename }}"
        passphrase: "{{ keyfile }}"
        state: opened
      # ansible.builtin.shell:
      #   cmd: "cryptsetup open --key-file - {{ item.device }} {{ item.mountpoint | basename }}"
      #   cmd: cat > test
      #   stdin: "{{ keyfile }}"
      #   stdin_add_newline: false
      loop: "{{ encrypted_mounts | dict2items(key_name='device', value_name='mountpoint') }}"
```

##### EXPECTED RESULTS

The encrypted partition is unlocked.

##### ACTUAL RESULTS

```paste below
TASK [unlock data partitions] ******************************************
failed: [b260-pc000] (item={'device': '/dev/vda3', 'mountpoint': '/mnt/data'}) => {"ansible_loop_var": "item", "changed": false, "item": {"device": "/dev/vda3", "mountpoint": "/mnt/data"}, "msg": "luks_device error: Error while opening LUKS container on /dev/vda3: No key available with this passphrase.\n"}
```

I've also tried reverting ansible/ansible@d0fda3e with the same result.

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
Fixes #827.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
luks_device

### CODE REVIEW

File: tests/sanity/ignore-2.10.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ plugins/modules/acme_account_info.py validate-modules:return-syntax-error
 plugins/modules/acme_challenge_cert_helper.py validate-modules:return-syntax-error
 plugins/modules/ecs_certificate.py validate-modules:invalid-documentation
 plugins/modules/get_certificate.py validate-modules:invalid-documentation
+plugins/modules/luks_device.py validate-modules:invalid-documentation
```
Comment (russoz) [inline tests/sanity/ignore-2.10.txt:12]: curiousity, what tripped the test? `text` as a value?

**Review by russoz — APPROVED** [review]
LGTM

File: tests/sanity/ignore-2.10.txt (hunk 1 above)
Comment (felixfontein) [reply tests/sanity/ignore-2.10.txt:12]: Nope, `choices` as a dictionary in `DOCUMENTATION`.

**Comment by felixfontein** [comment]
@russoz thanks for reviewing this!

File: tests/sanity/ignore-2.10.txt (hunk 1 above)
Comment (russoz) [reply tests/sanity/ignore-2.10.txt:12]: Why is that not allowed? It makes the docs better!

**Pull request merged**